commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -269,14 +268,12 @@ public static ExecutorService newFixedThreadPool(int numThreads, String name) {
                 SharedTaggedMetricRegistries.getSingleton(),
                 PTExecutors.wrap(
                         name,
-                        new AtlasRenamingExecutorService(
-                                ViewExecutor.builder(SHARED_EXECUTOR.get())
-                                        .setMaxSize(Math.min(numThreads, Short.MAX_VALUE))
-                                        .setQueueLimit(Integer.MAX_VALUE)
-                                        .setUncaughtHandler(AtlasUncaughtExceptionHandler.INSTANCE)
-                                        .build(),
-                                AtlasUncaughtExceptionHandler.INSTANCE,
-                                AtlasRenamingExecutorService.threadNameSupplier(name))),
+                        NylonExecutor.builder()
+                                .name(name)
+                                .executor(SHARED_EXECUTOR.get())
+                                .maxThreads(numThreads)
+                                .uncaughtExceptionHandler(AtlasUncaughtExceptionHandler.INSTANCE)
+                                .build()),
```

Comment:
the original code has some stuff around bounding the number of threads allowed to 32767: do we want to keep that / is there some protection at the layer underneath?

Reply:
In older versions of jboss-threads, the ViewExecutor had a max-value of `Short.MAX_VALUE`, beyond which it would throw a RuntimeException.
Not a concern here, if we get anywhere _close_ to 32k threads, the system will run out of resources...

Reply:
I see. Yeah to be fair 32k and infinity aren't that far off in this context :)